test_times.py

```diff
@@ -0,0 +1,17 @@
+from times import *
+
+def test_given_input():
+    large = time_range("2010-01-12 10:00:00", "2010-01-12 12:00:00")
+    short = time_range("2010-01-12 10:30:00", "2010-01-12 10:45:00", 2, 60)
+
+    result = large
+    expected = [('2010-01-12 10:00:00', '2010-01-12 12:00:00')]
+    assert result == expected
+    
+    result = short
+    expected = [('2010-01-12 10:30:00', '2010-01-12 10:37:00'), ('2010-01-12 10:38:00', '2010-01-12 10:45:00')]
+    assert result == expected
```

Comment:
I like that you test that `time_range` works as expected too, but I'd suggest splitting into different functions? A unit test should always only check one thing (a "unit").